internal/runners/export/deptree/artifacts.go

```diff
@@ -0,0 +1,142 @@
+package deptree
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/primer"
+	"github.com/ActiveState/cli/internal/runbits/rationalize"
+	"github.com/ActiveState/cli/internal/sliceutils"
+	"github.com/ActiveState/cli/pkg/buildplan"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/platform/model/buildplanner"
+	"github.com/ActiveState/cli/pkg/project"
+	"github.com/ActiveState/cli/pkg/sysinfo"
+	"github.com/go-openapi/strfmt"
+)
+
+type primeable interface {
+	primer.Auther
+	primer.Outputer
+	primer.Configurer
+	primer.Projecter
+	primer.Analyticer
+	primer.SvcModeler
+}
+
+type ArtifactParams struct {
+	Namespace  *project.Namespaced
+	CommitID   string
+	Req        string
+	PlatformID string
+	LevelLimit int
+}
+
+type DeptreeByArtifacts struct {
+	prime primeable
+}
+
+func NewByArtifacts(prime primeable) *DeptreeByArtifacts {
+	return &DeptreeByArtifacts{
+		prime: prime,
+	}
+}
+
+func (d *DeptreeByArtifacts) Run(params ArtifactParams) error {
+	logging.Debug("Execute DepTree")
+
+	out := d.prime.Output()
+	proj := d.prime.Project()
+	if proj == nil {
+		return rationalize.ErrNoProject
+	}
+
+	ns, err := resolveNamespace(params.Namespace, params.CommitID, d.prime)
+	if err != nil {
+		return errs.Wrap(err, "Could not resolve namespace")
+	}
+
+	bpm := buildplanner.NewBuildPlannerModel(d.prime.Auth())
+	commit, err := bpm.FetchCommit(*ns.CommitID, ns.Owner, ns.Project, nil)
+	if err != nil {
+		return errs.Wrap(err, "Could not get remote build expr and time for provided commit")
+	}
+
+	bp := commit.BuildPlan()
+
+	platformID := strfmt.UUID(params.PlatformID)
+	if platformID == "" {
+		platformID, err = model.FilterCurrentPlatform(sysinfo.OS().String(), bp.Platforms(), "")
+		if err != nil {
+			return errs.Wrap(err, "Could not get platform ID")
+		}
+	}
+
+	levelLimit := params.LevelLimit
+	if levelLimit == 0 {
+		levelLimit = 10
+	}
+
+	ingredients := bp.RequestedIngredients()
+	for _, ingredient := range ingredients {
+		if params.Req != "" && ingredient.Name != params.Req {
+			continue
+		}
+		out.Print(fmt.Sprintf("• [ACTIONABLE]%s/%s[/RESET] ([DISABLED]%s[/RESET])", ingredient.Namespace, ingredient.Name, ingredient.IngredientID))
+		d.printArtifacts(
+			nil,
+			ingredient.Artifacts.Filter(
+				buildplan.FilterPlatformArtifacts(platformID),
+			),
+			platformID,
+			1,
+			levelLimit,
+		)
+	}
+
+	return nil
+}
+
+func (d *DeptreeByArtifacts) printArtifacts(
+	parents []*buildplan.Artifact,
+	as buildplan.Artifacts,
+	platformID strfmt.UUID,
+	level int,
+	levelLimit int) {
+	indent := strings.Repeat("  ", level)
+	if level == levelLimit {
+		d.prime.Output().Print(indent + indentValue + "[ORANGE]Recursion limit reached[/RESET]")
```

Comment:
Nit: this isn't an error but a tip/note about passing the `--limit` flag to increase the recursion might helpful.